cpp/src/arrow/util/future.h

```diff
@@ -202,8 +202,30 @@ enum class FutureState : int8_t { PENDING, SUCCESS, FAILURE };
 
 inline bool IsFutureFinished(FutureState state) { return state != FutureState::PENDING; }
 
+/// \brief Describes whether the callback should be scheduled or run synchronously
+enum ShouldSchedule {
+  /// Always run the callback synchronously (the default)
+  NEVER = 0,
```

Comment:
We should avoid using ALL_CAPS names, because of potential clashes with macros (this is a common issue with Windows headers, unfortunately).

Reply:
Hmm, technically the style guide prefers `kAlways` but I see `Always` used more often in Arrow.  Although some of the gandiva code uses kAlways. (https://google.github.io/styleguide/cppguide.html#Enumerator_Names).  Any preference?

Reply:
`Always` sounds fine to me.

Reply:
Ok.  I'll make a PR to add this to the style guide docs as well.